Library/Homebrew/livecheck/strategy.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      module_function
+
+      def strategies
+        return @strategies if defined? @strategies
+
+        @strategies = {}
+        constants.sort.each do |strategy_symbol|
+          key = strategy_symbol.to_s.underscore.to_sym
+          strategy = const_get(strategy_symbol)
+          @strategies[key] = strategy
+        end
+        @strategies
+      end
+
+      def from_symbol(symbol)
+        strategies[symbol]
+      end
+
+      def from_url(url, regex_provided = nil)
+        usable_strategies = strategies.values.select do |strategy|
+          next if strategy.const_defined?(:PRIORITY) && !strategy::PRIORITY.positive?
+
+          strategy.respond_to?(:match?) && strategy.match?(url)
+        end
+
+        usable_strategies << strategies[:page_match] if strategies.key?(:page_match) && regex_provided
+
+        # Sort usable strategies in descending order by priority, using 5 as the
+        # default when a PRIORITY isn't provided in the Strategy itself.
+        usable_strategies.sort_by do |strategy|
+          (strategy.const_defined?(:PRIORITY) ? -strategy::PRIORITY : -5)
+        end
+      end
+    end
+  end
+end
+
+Dir["#{__dir__}/strategy/*.rb"].sort.each(&method(:require))
```

Comment:
I think it's worth making these explicit and perhaps e.g. adding a test if they aren't required.